services/brig/src/Brig/Sem/Now/MonadIO.hs

```diff
@@ -0,0 +1,41 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Brig.Sem.Now.MonadIO (brigTimeToMonadIO) where
+
+import Data.Time
+import Imports
+import Polysemy
+import Wire.Sem.FromUTC
+import Wire.Sem.Now
+
+-- | An interpreter of the 'Now' effect to MonadIO.
+brigTimeToMonadIO ::
```

Comment:
Is this interpretation different from the one already in Spar.Sem.Now.IO?

Reply:
Ignoring for a moment that the one you had in Spar runs in IO and this one runs in `m` for `MonadIO m`, this interpretation takes as input an IO action of type `IO UTCTime` from which it extracts the current time. After more thought in the afternoon, I realized that this interpretation is more general than the one in Spar/PR #2292 and the later can be implemented in terms of the former; I'll do that in PR #2292 and just reuse that more general interpreter in Brig.

Reply:
As I pointed out in PR #2292, I moved this function into the library as it is not Brig-specific. You can take a look there to see a subtle difference compared to how it used to be in Spar.Sem.Now.IO.